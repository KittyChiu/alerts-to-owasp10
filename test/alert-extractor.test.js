// alert-extractor.test.js
const extractAlerts = require('../src/alert-extractor')
const fs = require('fs').promises // Import the promises version of fs
require('dotenv').config()

describe('extractAlerts', () => {
  it('should extract alerts and match snapshot', async () => {
    const result = await extractAlerts(
      process.env.ORGANISATION,
      process.env.GITHUB_TOKEN
    )

    // Write the result to a file
    await fs.writeFile('./alerts.json', JSON.stringify(result, null, 2))

    expect(result).toMatchSnapshot()
  })
})
