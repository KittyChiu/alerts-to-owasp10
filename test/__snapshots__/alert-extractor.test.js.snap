// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractAlerts should extract alerts and match snapshot 1`] = `
{
  "webgoat-demo-2": {
    "1": [
      "cwe-079",
      "cwe-116",
    ],
    "10": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "11": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "12": [
      "cwe-020",
    ],
    "2": [
      "cwe-079",
      "cwe-116",
    ],
    "3": [
      "cwe-079",
      "cwe-116",
    ],
    "4": [
      "cwe-079",
      "cwe-116",
    ],
    "5": [
      "cwe-079",
      "cwe-116",
    ],
    "6": [
      "cwe-079",
      "cwe-116",
    ],
    "7": [
      "cwe-079",
      "cwe-116",
    ],
    "8": [
      "cwe-079",
      "cwe-116",
    ],
    "9": [
      "cwe-079",
      "cwe-116",
    ],
  },
  "webgoat-demo-mb": {
    "1": [
      "cwe-079",
      "cwe-116",
    ],
    "10": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "11": [
      "cwe-020",
      "cwe-080",
      "cwe-116",
    ],
    "2": [
      "cwe-079",
      "cwe-116",
    ],
    "3": [
      "cwe-079",
      "cwe-116",
    ],
    "4": [
      "cwe-079",
      "cwe-116",
    ],
    "5": [
      "cwe-079",
      "cwe-116",
    ],
    "6": [
      "cwe-079",
      "cwe-116",
    ],
    "7": [
      "cwe-079",
      "cwe-116",
    ],
    "8": [
      "cwe-079",
      "cwe-116",
    ],
    "9": [
      "cwe-079",
      "cwe-116",
    ],
  },
}
`;
